### PR TITLE
build: Add ASan Linux build to Actions build/test

### PR DESCRIPTION
**Pull request by VerteDinde**
#### Description of Change

This PR adds the Linux Address Sanitized (ASan) build to GitHub Actions. It also adds a verify check for mksnapshot and chromedriver to all builds outside of the ASan build.

#### Checklist

- [ ] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [ ] relevant documentation, tutorials, templates and examples are changed or added
- [ ] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: none

### CODE REVIEW

File: .github/workflows/pipeline-segment-electron-test.yml [hunk 1]
```diff
@@ -34,6 +39,7 @@ env:
   ELECTRON_OUT_DIR: Default
   ELECTRON_RBE_JWT: ${{ secrets.ELECTRON_RBE_JWT }}
   ELECTRON_GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  IS_ASAN: ${{ inputs.is-asan }}
```
Comment (MarshallOfSound) [inline .github/workflows/pipeline-segment-electron-test.yml:42]: does this have to be an env var? or can we use `${{ inputs.is-asan }}` directly instead? Env vars can make things hard to track where they're coming from, we should probably prefer to use inputs explicitly

Comment (codebytere) [reply .github/workflows/pipeline-segment-electron-test.yml:42]: Looks like we use this in a couple of tests:
- https://github.com/electron/electron/blob/040acaaf3026aec8148aa010283e650e629d682b/spec/asar-spec.ts#L117
- https://github.com/electron/electron/blob/a0a13ad623d2f0beae8b2574a22cf012100e6b6b/spec/api-app-spec.ts#L1488
- https://github.com/electron/electron/blob/67894f1493ca5bdbd5c3f22d6ea8f5db7f283950/spec/spellchecker-spec.ts#L16

We can probably get rid of it eventually but we probably do need it for now 🤔

File: .github/workflows/pipeline-segment-electron-test.yml [hunk 2]
```diff
@@ -159,7 +165,39 @@ jobs:
           chown -R :builduser . && chmod -R g+w .
           chmod 4755 ../out/Default/chrome-sandbox
           runuser -u builduser -- git config --global --add safe.directory $(pwd)
-          runuser -u builduser -- xvfb-run script/actions/run-tests.sh script/yarn test --runners=main --trace-uncaught --enable-logging --files $tests_files
+          if [ "$IS_ASAN" == "true" ]; then
```
Comment (codebytere) [inline .github/workflows/pipeline-segment-electron-test.yml:168]: Let's use inputs where possible?
```suggestion
          if [ "${{ inputs.is-asan }}" == "true" ]; then
```

File: .github/workflows/pipeline-segment-electron-test.yml [hunk 3]
```diff
@@ -159,7 +165,39 @@ jobs:
...
+            cd electron
+            runuser -u builduser -- xvfb-run script/actions/run-tests.sh script/yarn test --runners=main --trace-uncaught --enable-logging --files $tests_files | $ASAN_SYMBOLIZE
+          else
+            runuser -u builduser -- xvfb-run script/actions/run-tests.sh script/yarn test --runners=main --trace-uncaught --enable-logging --files $tests_files
+          fi
+        fi
+    - name: Verify mksnapshot
+      shell: bash
+      run: |
+        if [ "$IS_ASAN" != "true" ]; then
```
Comment (codebytere) [inline .github/workflows/pipeline-segment-electron-test.yml:187]:
```suggestion
        if [ "${{ inputs.is-asan }}" == "true" ]; then
```

File: .github/workflows/pipeline-segment-electron-test.yml [hunk 4]
```diff
@@ -159,7 +165,39 @@ jobs:
...
+          if [ "`uname`" != "Darwin" ] && ([ "$TARGET_ARCH" == "arm" ] || [ "$TARGET_ARCH" == "arm64" ]); then
+            python3 electron/script/verify-mksnapshot.py --source-root "$PWD" --build-dir out/Default --snapshot-files-dir $PWD/cross-arch-snapshots
+          else
+            python3 electron/script/verify-mksnapshot.py --source-root "$PWD" --build-dir out/Default
+          fi
+        fi
+    - name: Verify ChromeDriver
+      shell: bash
+      run: |
+        if [ "$IS_ASAN" != "true" ]; then
```
Comment (codebytere) [inline .github/workflows/pipeline-segment-electron-test.yml:198]:
```suggestion
        if [ "${{ inputs.is-asan }}" == "true" ]; then
```

File: .github/workflows/pipeline-segment-electron-build.yml [hunk 5]
```diff
@@ -63,6 +68,7 @@ env:
   CHECK_DIST_MANIFEST: true
   IS_GHA_RELEASE: true
   ELECTRON_OUT_DIR: Default
+  IS_ASAN: ${{ inputs.is-asan }}
```
Comment (codebytere) [inline .github/workflows/pipeline-segment-electron-build.yml:71]: We probably don't need this for build - it's only used in test!

File: .github/workflows/pipeline-segment-electron-test.yml (hunk 1 above)
Comment (MarshallOfSound) [reply .github/workflows/pipeline-segment-electron-test.yml:42]: Ah gotcha, can we use the input at least in the rest of the yml file?

**Comment by VerteDinde** [comment]
Green run here: https://github.com/electron/electron/actions/runs/9590522330/job/26446407462

**Review by MarshallOfSound — APPROVED** [review]

**Review by codebytere — APPROVED** [review]

**Pull request merged**